### PR TITLE
[Key Vault] Support API version 7.5-preview.1

### PR DESCRIPTION
**Pull request by mccoyp**
# Description

Resolves https://github.com/Azure/azure-sdk-for-python/issues/32101.

This adds support for the newest preview API version in each library, and also makes small API changes to Administration and Keys. The former library now has Managed Identity support for backup operations, and the latter has a new property on keys.

This PR also cleans up our generated code significantly by generating a single API version instead of every version we support. This redundancy turned out to be unnecessary, as a single generated client can send requests for any API version. This change also makes the `_generated_models.py` files in each library unnecessary; they were previously used to prevent references to a single API version's generated models.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by heaths** [comment]
Is there any way to separate the refactoring from the 7.5-preview.1 support? It's very hard to review this way. Can't tell what is new vs. what is refactoring.

**Comment by mccoyp** [comment]
> Is there any way to separate the refactoring from the 7.5-preview.1 support? It's very hard to review this way. Can't tell what is new vs. what is refactoring.

@heaths I could try to split things up if necessary, but the relevant changes only come from two commits:

- [`hsm_platform` support](https://github.com/Azure/azure-sdk-for-python/pull/31525/commits/f9e1f3d1437fa19dee4d37dfffee4c869f472f5a)
- [`use_managed_identity` support in backup/restore operations](https://github.com/Azure/azure-sdk-for-python/pull/31525/commits/3bddb9c073ef9c748c73a3b6a2e7d37cd6c30ee4)

If it's not too inconvenient, would reviewing based on those commits be possible? If not, I can try to split things into a regeneration PR and functionality PR.

**Review by heaths — APPROVED** [review]

**Review by heaths — APPROVED** [review]

**Comment by heaths** [comment]
> > Is there any way to separate the refactoring from the 7.5-preview.1 support? It's very hard to review this way. Can't tell what is new vs. what is refactoring.
> 
> @heaths I could try to split things up if necessary, but the relevant changes only come from two commits:
> 
> * [`hsm_platform` support](https://github.com/Azure/azure-sdk-for-python/pull/31525/commits/f9e1f3d1437fa19dee4d37dfffee4c869f472f5a)
> * [`use_managed_identity` support in backup/restore operations](https://github.com/Azure/azure-sdk-for-python/pull/31525/commits/3bddb9c073ef9c748c73a3b6a2e7d37cd6c30ee4)
> 
> If it's not too inconvenient, would reviewing based on those commits be possible? If not, I can try to split things into a regeneration PR and functionality PR.

I gave my approval - though I had a couple nits on one commit - based only on these commits. I'll leave it to @annatisch and/or @lmazuel or whoever about the other "refactorings".

File: sdk/keyvault/azure-keyvault-keys/CHANGELOG.md [hunk 1]
```diff
@@ -3,12 +3,14 @@
 ## 4.9.0b3 (Unreleased)
 
 ### Features Added
+- Added support for service API version `7.5-preview.1`
```
Comment (heaths) [inline sdk/keyvault/azure-keyvault-keys/CHANGELOG.md:6]: Add that the `hsm_platform` property was added?

File: sdk/keyvault/azure-keyvault-administration/CHANGELOG.md [hunk 2]
```diff
@@ -3,12 +3,14 @@
 ## 4.4.0b2 (Unreleased)
 
 ### Features Added
+- Added support for service API version `7.5-preview.1`
```
Comment (heaths) [inline sdk/keyvault/azure-keyvault-administration/CHANGELOG.md:6]: Added that `use_managed_identity` was added to back and restore?

**Review by heaths — APPROVED** [review]

File: sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_backup_client.py [hunk 3]
```diff
@@ -41,18 +43,37 @@ class KeyVaultBackupClient(KeyVaultClientBase):
...
-            https://<account>.blob.core.windows.net/backup
-        :param str sas_token: a Shared Access Signature (SAS) token authorizing access to the blob storage resource
+            https://<account>.blob.core.windows.net/backup.
 
-        :keyword str continuation_token: a continuation token to restart polling from a saved state
+        :keyword str sas_token: Optional Shared Access Signature (SAS) token to authorize access to the blob. Required
+            unless `use_managed_identity` is set to True.
+        :keyword use_managed_identity: Indicates which authentication method should be used. If set to True, Managed HSM
+            will use the configured user-assigned managed identity to authenticate with Azure Storage. Otherwise, a SAS
+            token has to be specified.
```
[the remark this answers is not in the document]
Comment (mccoyp) [reply sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_backup_client.py:74]: We just let the service handle it. Technically a user can also provide both a SAS token and `use_managed_identity=True`, but in that case we also just pass things along and leave it up to the service

File: sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_backup_client.py [hunk 4]
```diff
@@ -41,18 +43,37 @@ class KeyVaultBackupClient(KeyVaultClientBase):
         Vault or Managed HSM domain. Defaults to True.
     """
 
-    # pylint:disable=protected-access
-    @distributed_trace
+    @overload
+    def begin_backup(
```
[the remark this answers is not in the document]
Comment (heaths) [reply sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_backup_client.py:47]: FWIW, I see `restore_backup` is separated across newlines, but it has 1 additional parameter so maybe that's the proverbial straw.

Note, this is just a nit. If it's the formatter doing this - or you just don't feel like it :) - feel free to disregard.

[the remark this answers is not in the document]
Comment (mccoyp) [reply sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_backup_client.py:47]: This is just down to `black` 🙂 We keep things on one line if it doesn't exceed 120 columns, but we can also split them into newlines. I might do that to align across overloads and to make the differences between them more obvious

**Pull request merged**